clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventHandler.java

```diff
@@ -73,7 +90,9 @@ public ApplicationEventHandler(final LogContext logContext,
      */
     public void add(final ApplicationEvent event) {
         Objects.requireNonNull(event, "ApplicationEvent provided to add must be non-null");
+        event.setAddedToQueueMs(System.currentTimeMillis());
```

Comment:
This needs to get the current time in milliseconds from the `Time` object that was passed in to the constructor.